lib/pure/collections/sequtils.nim

```diff
@@ -706,7 +706,7 @@ template newSeqWith*(len: int, init: untyped): untyped =
     result[i] = init
   result
 
-macro asArray*(targetType: untyped, values: typed): untyped =
+macro asArray*(targetType: typedesc, values: typed): untyped =
```

Comment:
@Araq, yes that was obviously my first attempt as well. However, this does not work. When I attempt to add `targetType` to the result NimNode, i get the following error from the compiler:
> sequtils.nim(727, 9) Error: type mismatch: got (NimNode, typedesc)

The only way, I can see, to fix this is to use `getType` on `targetType` to get the symbol that is bound to the target type. However, you did not want me to do that.

Reply:
I have a solution for this that I want to try out.